Belindas-closet-nextjs-4-41-add-a-product-functionality

taylorpapke opened the pull request:
improves upon: https://github.com/SeattleColleges/belindas-closet-nextjs/issues/41

completed:
- added another field for description
- upon submit, both fields are converted to JSON format which can then be viewed in browser console (doesn't write to JSON file!)

other notes:
- tried using postman to see if I could GET from the URL and see the JSON data but ran into errors with NextAuth.js. I gave up on this method (for the moment) after moving files around and adding more code which seemed to overcomplicate matters.
- tried to write to a temp json file but ran into errors using fs module. I gave up on this method (for the moment) because importing more libraries and adding more code didn't seem like the best option.

questions I have:
- the purpose of all this is to try to build the frontend UI without having the backend available so that it can be integrated smoothly when it is. Does anyone have any other ideas of how this could be accomplished?

Seiyaroo commented:
You should still be able to get a local-run version of things using react without needing any backend connections I am pretty sure?   But maybe that's not what you're asking / intending

brinkbrink commented:
I think I need more clarity on what you are testing. You may be testing outside the scope of what is needed for this feature, since it is just the UI.

A quote from [this article](https://www.testim.io/blog/user-interface-testing/) says it well, "As a rule of thumb, remember that UI testing should worry about the look and behavior of the UI. How the visual elements are presented, how they interact, respond to user input, and validate input data. That should be the scope of UI testing." 

The whole article that is linked from is pretty helpful for understanding the context of UI testing.

taylorpapke commented:
Yes @brinkbrink, perhaps this PR is overdoing it a bit since it is just UI overall (thanks for pointing that out). My only concern was adding too many fields for fear of later trying to submit many different data points and what trouble could occur. I think this is where I went off on a tangent of trying to validate that the data was actually being submitted while the fields were limited which is clearly outside the scope of adding simple product functionality UI. I will fix and make another push to this PR.

taylorpapke commented:
Updated PR with the following changes:
- removed console.logs for inputed data as strings in .json format (outside scope of UI).
- added drop down menu for size.
- changed product type text box to a drop down menu to stay more in line with the format of the inventory submission google form.

theGaryLarson commented on app/api/auth/product-page/page.tsx:
```diff
@@ -4,14 +4,25 @@ import React, { useState } from 'react';
+    setProductDescription(e.target.value);
   }
+  
+  const handleProductSelect = (e: { target: { value: React.SetStateAction<string>; }; }) => {
```
Consider using more descriptive variable names for event handlers to improve code readability. For example, use handleProductChange instead of handleProductSelect. Grouping similar state-related functions together can also enhance code organization.

theGaryLarson commented on app/api/auth/product-page/page.tsx:
```diff
@@ -4,14 +4,25 @@ import React, { useState } from 'react';
+    setSizeValue(e.target.value);
+  };
+  
   const handleSubmit = (e: any) => {
```
Like your style! Iterative and stubbing out that code!

theGaryLarson commented on app/api/auth/product-page/page.tsx:
```diff
@@ -20,8 +31,46 @@ const AddProduct = () => {
+          </div>
+          <div>
+          <label>Size:</label>
+            <select value={sizeValue} onChange={handleSizeSelect} style={{color: 'black'}}>
```
A recommendation but it could be a refactor later as well.
You could use CSS classes or styled-components to handle styling for better separation of concerns

brinkbrink reviewed (approved):
Great progress! Thank you, Taylor!
